File: RegDFA/src/Edge.cpp
#include <string>
#include <iostream>
#include "Edge.h"
using namespace std;

Edge::Edge(Vertex* vertT, char weigh){
	weight = weigh;
	vertexTo=vertT;
	//dangling =NULL;

}
//Copy Constructor
Edge::Edge(const Edge &E){
	vertexTo = E.vertexTo;
	weight = E.weight;
	//dangling =NULL;
}


void Edge::makeFinalFalse(){
	vertexTo->thisState.finalState=false;
}
void Edge::makeFinalTrue(){
	vertexTo->thisState.finalState=true;
}

void Edge::makeRootFalse(){
	vertexTo->thisState.rootState=false;
}
void Edge::makeRootTrue(){
	vertexTo->thisState.rootState=true;
}

Edge& Edge::operator=(const Edge& T){
	vertexTo = T.vertexTo;
	weight = T.weight;
	//dangling =T.dangling;
	return *this;
}

File: RegToDFA/src/Edge.java


public class Edge{
	private char weight;
	private Vertex vertexTo;
	
	
	public Edge(Vertex vertT, char weigh){
		weight = weigh;
		vertexTo = vertT;
	}
		
	public Vertex returnToVertex(){
		return this.vertexTo;
	}
	public char returnWeight(){
		return weight;
	}
	

	
	
	
	
}
